### ISSUE TITLE
MBTiles cache databases have their "metadata" table unpopulated

### ISSUE THREAD

**Issue by dalbani** [mapproxy/mapproxy#64]
While trying to use the `mbcheck` program (from the `node-mbtiles` package), I've discovered that the `metadata` table is empty in the MBTiles databases produced by MapProxy.

There is indeed a method called `update_metadata()` in `cache/mbtiles.py` but it doesn't seem to be called.
Furthermore, it doesn't support the `bounds` property, which is recommended by the MBTiles spec.

Producing fully populated MBTiles databases would help when working with third-party tools.

**Comment by PowerPan** [comment]
ok bring this issue back to the top now ;)

The specification is:
https://github.com/mapbox/mbtiles-spec/blob/master/1.1/spec.md

I have here a screenshot from a other file this should help:

![mbtilesmetadata](https://cloud.githubusercontent.com/assets/3910878/25953382/73e88f8c-3663-11e7-9e37-56df02998ccb.png)

And the "Demo" Data
[10.mbtiles.zip](https://github.com/mapproxy/mapproxy/files/993230/10.mbtiles.zip)

@olt I Think this should help

**Comment by olt** [comment]
> It's not clear to me what metadata MapProxy should write into the MBTiles cache, since a cache normally doesn't now the bounds or the title/name.

A cache grows over time and it would be inefficient to recalculate and update the bounding box with every stored tile. That's what I meant with _a cache normally doesn't now the bounds_.

**Comment by PowerPan** [comment]
yes that right, I think the main problem is that the table is missing.
I get an error that it is missing when I try to open the mbtiles file in MapTiler

![bildschirmfoto 2017-05-11 um 16 55 40](https://cloud.githubusercontent.com/assets/3910878/25955851/9973f708-366a-11e7-80b8-67dd5c2278c2.png)

maybe for the moment to have this table enter the whole range
bounds: -180,-90,180,90
center: 0,0

that is not the best solution I know but maybe later we can found a better way

**Comment by PowerPan** [comment]
@olt I am agree with you that it is "nearly" impossible for the moment for the cache to define a boundix-box using the normal mapproxy

but when I using mapproxy-seed than it should be possible to set this metadata at the end of the seeding process, isn't it?

**Comment by ache051** [comment]
Hate to resurrect an old thread. Tileserver-GL has been recognizing MBTiles created by MapProxy as vector MBTiles. After some frustrating efforts including rebuilding indices and jigging table structures, I found that it is due to the empty metadata table! Once I do "INSERT INTO metadata (name, value) VALUES('format','jpg');" Tileserver-GL immediately recognize it as raster MBTiles and renders tiles perfectly. Is it worth any effort to populate the metadata table when the MBTiles file is created?

**Comment by drnextgis** [comment]
There is a [method](https://github.com/mapproxy/mapproxy/blob/53745e70b628b20c172709fd1503b67dea2b27d0/mapproxy/cache/mbtiles.py#L111)  for doing that but I don't see it being called anywhere.
